### PR TITLE
stackrox: Add support for OCP next e2e test runs

### PR DESCRIPTION
**Pull request by gavin-stackrox**
With Openshift CI we can get access to 4-dev-preview clusters which will allow us to test against new OCP versions prior to release. This PR adds `/test` hooks for each of the three OCP e2e test flavors. The fact that they are failing is to be expected.

/uncc
/hold

### CODE REVIEW

**Comment by gavin-stackrox** [comment]
/pj-rehearse pull-ci-stackrox-stackrox-master-ocp-next-qa-e2e-tests

**Comment by gavin-stackrox** [comment]
/pj-rehearse pull-ci-stackrox-stackrox-master-ocp-next-qa-e2e-tests

**Comment by gavin-stackrox** [comment]
/pj-rehearse pull-ci-stackrox-stackrox-master-ocp-next-ui-e2e-tests pull-ci-stackrox-stackrox-master-ocp-next-qa-e2e-tests pull-ci-stackrox-stackrox-master-ocp-next-operator-e2e-tests

**Review by porridge — APPROVED** [review]
I don't know what to think about the `job-release: "4.13"` label that got added, but the goal is noble.

**Comment by porridge** [comment]
This is interesting:
```
  Message:  couldn't ensure registry server - error ensuring pod: : error creating new pod: stackrox-operator-test-index-: pods "stackrox-operator-test-index-56h6d" is forbidden:
 violates PodSecurity "restricted:v1.24":
 allowPrivilegeEscalation != false (container "registry-server" must set securityContext.allowPrivilegeEscalation=false),
 unrestricted capabilities (container "registry-server" must set securityContext.capabilities.drop=["ALL"]),
 runAsNonRoot != true (pod or container "registry-server" must set securityContext.runAsNonRoot=true),
 seccompProfile (pod or container "registry-server" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
```

Let me file a ticket about this...

File: ci-operator/config/stackrox/stackrox/stackrox-stackrox-master.yaml [hunk 1]
```diff
@@ -24,6 +24,19 @@ build_root:
...
+      product: ocp
+      stream: 4-dev-preview
+      version_bounds:
+        lower: 4.1.0
+        upper: 4.999.0
+  latest:
+    candidate:
+      product: ocp
+      stream: ci
+      version: "4.13"
```
Comment (msugakov) [inline ci-operator/config/stackrox/stackrox/stackrox-stackrox-master.yaml:39]: This seems to be unused or does it play some role?

File: ci-operator/config/stackrox/stackrox/stackrox-stackrox-master.yaml [hunk 2]
```diff
@@ -319,6 +332,39 @@ tests:
   steps:
     workflow: stackrox-automation-flavors-rosa-e2e
   timeout: 5h0m0s
+- as: ocp-next-qa-e2e-tests
+  steps:
+    cluster_profile: aws
+    dependencies:
+      OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: release:ec
+    env:
+      TEST_SUITE: openshift-newest-qa-e2e-tests
```
Comment (msugakov) [inline ci-operator/config/stackrox/stackrox/stackrox-stackrox-master.yaml:341]: I wonder why did we have to create symlinks in `stackrox/stackrox` if we could just use this to override job name.

File: ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-master-postsubmits.yaml [hunk 3]
```diff
@@ -10,6 +10,7 @@ postsubmits:
       skip_cloning: true
     labels:
       ci.openshift.io/generator: prowgen
+      job-release: "4.13"
```
Comment (msugakov) [inline ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-master-postsubmits.yaml:13]: Did you introduce changes to these files manually or were they autogenerated?

File: ci-operator/config/stackrox/stackrox/stackrox-stackrox-master.yaml (hunk 1 above)
Comment (gavin-stackrox) [reply ci-operator/config/stackrox/stackrox/stackrox-stackrox-master.yaml:39]: This is required to get the job to run. I don't really understand the `releases` stanza and _should_ probably read the docs more.

File: ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-master-postsubmits.yaml (hunk 3 above)
Comment (gavin-stackrox) [reply ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-master-postsubmits.yaml:13]: Autogenerated. And goes to my comment on the 'releases' stanza earlier.

File: ci-operator/config/stackrox/stackrox/stackrox-stackrox-master.yaml [hunk 4]
```diff
@@ -319,6 +332,39 @@ tests:
   steps:
     workflow: stackrox-automation-flavors-rosa-e2e
   timeout: 5h0m0s
+- as: ocp-next-qa-e2e-tests
```
Comment (msugakov) [inline ci-operator/config/stackrox/stackrox/stackrox-stackrox-master.yaml:335]: Please don't forget `optional: true` on these `ocp-next` steps.

**Review by msugakov — APPROVED** [review]

**Comment by gavin-stackrox** [comment]
Thanks for spotting the missing `optional:` annotation. This means this will need another `/lgtm` though.
/cc @msugakov @porridge 
/pj-rehearse ack

**Review by porridge — APPROVED** [review]
/lgtm

**Comment by gavin-stackrox** [comment]
I had to fix conflicts and so this will need another /lgtm. Some new lines got added to `ci-operator/jobs` but as that was done by a rebased `make update`, I am going to assume that the automation knows what it is doing.
/cc @msugakov @porridge
/pj-rehearse ack

**Review by BradLugo — APPROVED** [review]

**Pull request merged**